### PR TITLE
Add problem matchers to highlight errors in Github PRs code diff

### PR DESCRIPTION
**Pull request by armab**
Based on the previous PRs, sometimes the linting error messages are not clear enough.
Add problem matches to Github Actions so we can highlight the warnings and errors directly in the *PR code diff* for specific `file:line` from `pylint` and `black` CI runs.

See: https://github.com/actions/toolkit/blob/master/docs/commands.md#problem-matchers
See: https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md

### The Result
Pylint:
image

Black:
image

So there's no need to go into the GH Action Details URL, as the result is just ^^ there.

This PR also splits `black` vs `pylint` in the GH Actions so the failure is more clear:
image

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -25,15 +26,42 @@ jobs:
           path: |
             ~/.cache/pip
             .tox
-          key: ${{ runner.os }}-v1-python-3.8-${{ hashFiles('requirements.txt', 'test-requirements.txt', 'pyproject.toml', 'tox.ini') }}
+          key: ${{ runner.os }}-v1-python-3.8-black-${{ hashFiles('test-requirements.txt', 'pyproject.toml', 'tox.ini') }}
```
Comment (PhillypHenning) [inline .github/workflows/ci.yaml:29]: `test-requirements.txt` -> `requirements.txt`

File: scripts/plugins/install_plugins.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import os
 import git
 import yaml
+import non_existent_module
```
Comment (PhillypHenning) [inline scripts/plugins/install_plugins.py:8]: Is this just to illustrate that the UI formatting works? 
Either way this should be removed before merge

Comment (armab) [reply scripts/plugins/install_plugins.py:8]: Right, that's for illustration for you folks to get an idea of how it looks like in the PR code diff.

Will remove it before merging and also plan to squash all the commits as there's a lot of commit noise in this PR.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (armab) [reply .github/workflows/ci.yaml:29]: No that's as expected. `back` depends only on the `test-requirements.txt`, - that's where we define it as a test/dev dependency.

Comment (armab) [reply .github/workflows/ci.yaml:29]: `pylint` (below) also needs `requirements.txt` as it does more in-depth code analysis for the project, including its pip  dependencies.

**Pull request merged**